### PR TITLE
user authorization for edit/update & destroy review

### PR DESCRIPTION
**Pull request by detung**

### CODE REVIEW

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -43,14 +43,24 @@ def destroy
...
+
     flash[:notice] = 'Review deleted'
 
     redirect_to superpower_path(@superpower)
-
   end
 
   private
 
+  def authorize_user
```
Comment (bkincart) [inline app/controllers/reviews_controller.rb]: It doesn't look like we're actually using this method at all, please either implement or remove!

File: spec/features/admin_update_and_delete_superpowers_spec.rb [hunk 2]
```diff
@@ -24,10 +23,10 @@
     expect(page).to have_content("Superpower successfully updated")
   end
 
-  scenario 'user should only see edit link if they added the superpower' do
+  scenario 'only admin should see edit link' do
     user2 = FactoryBot.create(:user)
     sign_in_as(user2)
-    user1 = FactoryBot.create(:user)
+    user1 = FactoryBot.create(:user, role: 'admin')
```
Comment (bkincart) [inline spec/features/admin_update_and_delete_superpowers_spec.rb]: I'd really like to clean this up by declaring `user1` above `user2` (in other words, just re-ordering your lines)

File: spec/features/admin_add_superpowers_spec.rb [hunk 3]
```diff
@@ -1,9 +1,9 @@
 require 'rails_helper'
 
-feature 'user can add superpowers' do
+feature 'admin can add superpowers' do
```
Comment (bkincart) [inline spec/features/admin_add_superpowers_spec.rb]: Do you ONLY want admins to be able to create superpowers?

If all users should be able to create superpowers, you definitely want to remove the "admin" aspect from this test to make sure all users can create.

File: app/views/admin/users/index.html.erb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <h1>Registered Users</h1>
 
   <% @users.each do |user| %>
-    <% if user.role == "member" %>
+    <% if user.member? %>
```
Comment (bkincart) [inline app/views/admin/users/index.html.erb]: Nice refactor!

**Review by bkincart — CHANGES_REQUESTED** [review]

File: spec/features/admin_add_superpowers_spec.rb (hunk 3 above)
Comment (detung) [reply spec/features/admin_add_superpowers_spec.rb]: Yes, only admins can create superpowers

File: app/controllers/reviews_controller.rb (hunk 1 above)
Comment (detung) [reply app/controllers/reviews_controller.rb]: will implement in a `before_action` call

**Review by bkincart — APPROVED** [review]

**Pull request merged**